packages/immutable-arraybuffer/index.js

```
@@ -0,0 +1,152 @@
+    return this.#buffer.byteLength;
+  }
+
+  get detached() {
+    this.#buffer; // shim brand check
+    return false;
+  }
+
+  get maxByteLength() {
+    // Not underlying maxByteLength, which is irrelevant
+    return this.#buffer.byteLength;
+  }
+
+  get resizable() {
+    this.#buffer; // shim brand check
+    return false;
+  }
+
+  get immutable() {
+    this.#buffer; // shim brand check
+    return true;
+  }
+
+  slice(begin = 0, end = undefined) {
+    return arrayBufferSlice(this.#buffer, begin, end);
+  }
+
+  resize(_newByteLength = undefined) {
+    this.#buffer; // shim brand check
+    throw TypeError('Cannot resize an immutable ArrayBuffer');
+  }
+
+  transfer(_newLength = undefined) {
+    this.#buffer; // shim brand check
+    throw TypeError('Cannot detach an immutable ArrayBuffer');
+  }
+
+  transferToFixedLength(_newLength = undefined) {
+    this.#buffer; // shim brand check
+    throw TypeError('Cannot detach an immutable ArrayBuffer');
+  }
+
+  transferToImmutable(_newLength = undefined) {
+    this.#buffer; // shim brand check
+    throw TypeError('Cannot detach an immutable ArrayBuffer');
+  }
+}
+
+const immutableArrayBufferPrototype = ImmutableArrayBufferInternal.prototype;
+// @ts-expect-error can only delete optionals
+delete immutableArrayBufferPrototype.constructor;
+
+const {
+  // @ts-expect-error We know it is there.
+  get: isImmutableGetter,
+} = getOwnPropertyDescriptor(immutableArrayBufferPrototype, 'immutable');
+
+setPrototypeOf(immutableArrayBufferPrototype, arrayBufferPrototype);
+
+export const transferBufferToImmutable = buffer =>
+  new ImmutableArrayBufferInternal(buffer);
+
+export const isBufferImmutable = buffer => {
+  try {
+    return apply(isImmutableGetter, buffer, []);
+  } catch (err) {
+    if (err instanceof TypeError) {
+      // TODO: Should this test if `buffer` is a genuine `ArrayBuffer` exotic
+      // object, and throw if not?
+      return false;
+    }
+    throw err;
+  }
```

Looking at existing getters such as [`resizable`](https://tc39.es/ecma262/multipage/structured-data.html#sec-get-arraybuffer.prototype.resizable), we should throw on invalid input.
```suggestion
  try {
    return apply(isImmutableGetter, buffer, []);
  } catch (err) {
    if (err instanceof TypeError) {
      // Enforce that `buffer` is a genuine ArrayBuffer before returning.
      arrayBufferSlice(arrayBuffer, 0, 0);
      return false;
    }
    throw err;
  }
```

---

Done, thanks.